arch/x86/soc/intel_quark/quark_se/soc.c

```diff
@@ -21,6 +21,34 @@
 #include <uart.h>
 #include <init.h>
 #include "shared_mem.h"
+#include <mmustructs.h>
+
+
+#ifdef CONFIG_X86_MMU
+MMU_BOOT_REGION(CONFIG_PHYS_LOAD_ADDR, CONFIG_ROM_SIZE*1024, MMU_ENTRY_WRITE);
+
+MMU_BOOT_REGION(CONFIG_PHYS_RAM_ADDR, CONFIG_RAM_SIZE*1024, MMU_ENTRY_WRITE);
+
+/* loapic */
+MMU_BOOT_REGION(CONFIG_LOAPIC_BASE_ADDRESS, 4*1024, MMU_ENTRY_WRITE);
```

Comment:
I took these values from the datasheet for quark c1000.
[http://www.intel.com/content/dam/www/public/us/en/documents/datasheets/quark-c1000-datasheet.pdf](http://www.intel.com/content/dam/www/public/us/en/documents/datasheets/quark-c1000-datasheet.pdf)